### PR TITLE
session 1: notes + suggestions

### PR DESCRIPTION
**Pull request by smhutch**
Hey @ThomasHoadley, 

Here's the notes changes I made during our session.
Thanks!

### CODE REVIEW

File: client/src/components/List.jsx [hunk 1]
```diff
@@ -56,17 +30,39 @@ class List extends Component {
...
+	const { id } = ownProps.match.params
 	const { lists } = state;
-	return { lists };
+
+	const list = lists.filter(list => {
+		console.log(list)
+		return list.id === id
+	})
+
+	console.log(list)
```
Comment (smhutch) [inline client/src/components/List.jsx:51]: As we found during the call, something wasn't working quite right here. An issue with doing things under a time pressure 😆 

The purpose of what we spoke about here was to transform the data from the Redux store, and only pass what we need into the component.

File: client/src/components/Lists.jsx [hunk 2]
```diff
@@ -8,16 +8,18 @@ class Lists extends Component {
 		super(props);
 		this.handleAddList = this.handleAddList.bind(this);
 		this.renderLists = this.renderLists.bind(this);
-		this.handleDelete = this.handleDelete.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
 	}
 
+  // Don't be afraid of inline event handlers
```
Comment (smhutch) [inline client/src/components/Lists.jsx:14]: I.e. passing a function directly as a prop


```tsx
<Component
  onClick={() => {
    // do something here.
  }}
/>
```

File: client/src/components/Lists.jsx [hunk 3]
```diff
@@ -26,12 +28,16 @@ class Lists extends Component {
 	}
 
 	renderLists() {
-		const lists = [];
+    const lists = [];
 		this.props.lists.map((list) => {
-			let { title, id } = list;
+      let { title, id } = list;
+      // TODO: return directly, instead of mapping.
```
Comment (smhutch) [inline client/src/components/Lists.jsx:34]: i.e.

```tsx
this.props.list((list) => (
  <div className="listItem"
...
</div>
)}
```

File: client/src/components/Lists.jsx [hunk 4]
```diff
@@ -26,12 +28,16 @@ class Lists extends Component {
...
 			lists.push(
+        // TODO: unique keys
+        // https://reactjs.org/docs/lists-and-keys.html
 				<div className="list-item">
 					<Link to={`/list/${id}`}>{title}</Link>
+          {/* TODO: use a button */}
```
Comment (smhutch) [inline client/src/components/Lists.jsx:40]: This is an **a11y** suggestion, not specific to React

File: client/src/store/reducers.js [hunk 5]
```diff
@@ -2,15 +2,12 @@ import { ADD_LIST, DELETE_LIST, ADD_LIST_ITEM } from "./actions";
 import { v4 as uuidv4 } from "uuid";
 
 const initialState = {
-	lists: [
+	// Consider using a map here
```
Comment (smhutch) [inline client/src/store/reducers.js:5]: i.e. an object with the resource id as it's key

File: client/src/store/reducers.js [hunk 6]
```diff
@@ -60,11 +63,15 @@ function listsApp(state = initialState, action) {
...
-			});
+			return {
+				...state,
+				lists: state.lists.filter((list) => list.id !== action.id)
+			}
+			// return Object.assign({}, state, {
+			// 	lists: state.lists.filter((list) => {
+			// 		return list.id !== action.id;
+			// 	}),
+			// });
```
Comment (smhutch) [inline client/src/store/reducers.js:74]: Just another way of doing the same thing. It's mostly personal preference if you use `...spread` or `Object.assign` 👍

**Comment by ThomasHoadley** [comment]
Thanks again Scott, I've pulled the branch down onto a seperate local and am refactoring now, starting without Redux!

File: client/src/components/Lists.jsx (hunk 3 above)
Comment (ThomasHoadley) [reply client/src/components/Lists.jsx:34]: May need quick clarification on this for next session

**Pull request closed without merging**